metal3-templater.go

```diff
@@ -64,6 +70,14 @@ func main() {
 	templateData.ClusterIronicURL = fmt.Sprintf("http://%s", net.JoinHostPort(ip, "6385"))
 	templateData.ClusterInspectorURL = fmt.Sprintf("http://%s", net.JoinHostPort(ip, "5050"))
 
+	// http_basic credentials
+	if numArgs == 11 {
+		templateData.IronicUser = fmt.Sprintf("%s", os.Args[7])
+		templateData.IronicPassword = fmt.Sprintf("%s", os.Args[8])
```

Comment:
I'd suggest combining username and password into a single argument as `<username>:<password>`.

Reply:
not sure if it would be worth having them in a single argument and than we need to split..

Reply:
You always have to pass username and password together, so it makes sense to have a single argument like e.g. `--ironic-basic-auth=<user>:<pass>`.
The semantics are well-defined because it's traditional to pass basic auth creds in a url like `http://<user>:<pass>@<host>/<path>`, and the only reason we're not doing that here is that we're passing only the cluster IP and constructing the URLs ourselves because the IP is used in multiple places.

Reply:
this would be for the CLI, but we need to have different variables or we will need to add things in the `clouds.yaml.template` to split `<user>:<pass>` and put in the correct field no?

Reply:
Yeah, I meant combine them in the CLI and split them in code here.

Reply:
I've updated the code and I will let run to see how it goes using <user>:<password>, tomorrow I will push a new commit in this PR